File: conversational-hub-flutter/lib/features/chat/domain/entities/channel.dart
import 'package:equatable/equatable.dart';

class Channel extends Equatable {
  final String id;
  final String name;
  final String? description;
  final String? erpEntityId;
  final String? erpEntityType;
  final DateTime createdAt;

  const Channel({
    required this.id,
    required this.name,
    this.description,
    this.erpEntityId,
    this.erpEntityType,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, name, description, erpEntityId, erpEntityType, createdAt];
}

File: conversational-hub/src/main/java/com/allianceever/conversationalhub/domain/entities/Channel.java
package com.allianceever.conversationalhub.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Channel {
    private String id;
    private String tenantId;
    private String name;
    private String description;
    private String erpEntityId;
    private String erpEntityType;
    private LocalDateTime createdAt;
    private boolean isAutomatic;
}
